src/dotnet/CarbonAware/ICarbonAware.cs

```diff
@@ -9,13 +9,6 @@ public interface ICarbonAware
     /// </summary>
     /// <param name="props">IDictionary with properties required by concrete classes</param>
     /// <returns>An IEnumerable instance with EmissionsData instances.</returns>
-    Task<IEnumerable<EmissionsData>> GetEmissionsDataAsync(IDictionary props);
+    Task<IEnumerable<EmissionsData>> GetEmissionsDataAsync(Dictionary<string, object> props);
```

Comment:
@juzuluag this is something Priti & I thought you should specifically look at. We changed it to a Dictionary type so you can do key validation easily. If we keep as an IDictionary then I think we would need to wrap each key access in a try/catch and then do the proper parsing/casting once we know it's not null. Thoughts?

Reply:
@JenMadiedo , understand, but having it as an iterface, and defining the keys as part of the `CarbonAwareContants` should be fine to do for instance:
`props[ContantsKeys.Time]`